sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/sonar/TableWithoutHeaderCheck.java

```diff
@@ -37,12 +38,23 @@ private static boolean isTable(TagNode node) {
 
   private static boolean isLayout(TagNode node) {
     String role = node.getAttribute("role");
-    return role != null && ("PRESENTATION".equalsIgnoreCase(role) || "NONE".equalsIgnoreCase(role));
+    return ("PRESENTATION".equalsIgnoreCase(role) || "NONE".equalsIgnoreCase(role));
   }
 
   private static boolean isHidden(TagNode node) {
     String ariaHidden = node.getAttribute("aria-hidden");
-    return ariaHidden != null && "TRUE".equalsIgnoreCase(ariaHidden);
+    return "TRUE".equalsIgnoreCase(ariaHidden);
+  }
+
+  private static boolean hasThymeleafTemplate(TagNode node) {
+    return node.getAttributes().stream().anyMatch(TableWithoutHeaderCheck::isAttributeThymeleafTemplated) // Thymeleaf in table attributes ?
+      || node.getChildren().stream().map(TagNode::getAttributes).anyMatch(                                // Thymeleaf in children attributes ?
```

Comment:
Thanks for spotting this. Here is my consideration on this:
- If `th:insert` or `th:include` occur as an attribute of `<table>`, they might insert a TH as a child of `<table>` so compliant code.
- If `th:replace` occurs as an attribute of `<table>`, it is out of scope because it will remove`<table>`. I supposed, maybe wrongly,  we shouldn't encounter such code.
- If `th:block` is a child of `<table>` and has `th:replace,th:insert,th:include`, I assumed a TH could be inserted, so it is compliant. Remark that I'm not sure of the behavior of `th:replace` in `th:block`, but I think it is good enough like this.
- If `th:replace` in any child, it could remove the child tag and replace it with a TH, so it is compliant.

Let me know if you agree with this and if this is sufficiently clear in the code

Reply:
After consideration and discussion, it becomes difficult to split the logic between `th:replace` and `th:insert`.It would require to identify each tag name which can include a `<th></th>` in a proper way. 

However, `TableWithoutHeaderCheck` is not that smart for non-template HTML : The code is compliant as soon as it finds a `<th></th>`. I don't think doing something more imaginative for thymeleaf templates is worth it. 

So I simplified the logic : If we insert a thymeleaf fragment using `th:replace` or `th:insert` or `th:include` anywhere inside the hierarchy, I assume the code is compliant.